src/Models/NovaSentMail.php

```diff
@@ -83,4 +67,18 @@ public function setMailTemplateAttribute($value)
         $value = $value instanceof NovaMailTemplate ? $value->id : $value;
         $this->attributes['mail_template_id'] = $value;
     }
+
+    /**
+     * The "booting" method of the model.
+     *
+     * @return void
+     */
+    protected static function boot()
+    {
+        parent::boot();
+
+        static::creating(function ($mail) {
```

Comment:
What if there is no authenticated user and the event was triggered by an update outside the browser request cycle?